### ISSUE TITLE
Connecting my wp.com account fails in FF with strict tracking protection

### ISSUE THREAD

**Issue by jsnajdr** [Automattic/jetpack#19878]
#### Steps to reproduce the issue

1. Use Firefox with Enhanced Tracking Protection set to "strict"
2. Create a jurassic.ninja site and try to connect Jetpack to your wp.com account
3. After logging-in in a popup, I eventually got to this screen in the flow:

<img width="504" alt="Screenshot 2021-05-18 at 11 19 07" src="https://user-images.githubusercontent.com/664258/118631649-02069b00-b7d0-11eb-9e21-5c036b7f5e75.png">

Clicking the "Approve" button failed to connect me and I can't make further progress.

Some analysis of what's going on:
The "connect as ..." box is an iframe loaded from `https://jetpack.wordpress.com/jetpack.authorize_iframe`. That iframe, although it's cross-origin (jurassic.ninja site loads wordpress.com frame), is loaded with the `wordpress_logged_in` cookie, so the frame knows who I am. The request and response also contain a `wpcom3rdParty` cookie. So, if Jetpack tried to detect blocked 3rd party cookies here, the detection failed and concluded that 3rd party cookies are OK.

Then the iframe loads a nested iframe: the REST proxy. Now on _this_ iframe request the 3rd party login cookies (`wordpress_sec` and `wordpress_logged_in`) are blocked. The iframe is not authorized and the REST request that is supposed to connect the site:
```
https://public-api.wordpress.com/rest/v1/jetpack-blogs/:site/jetpack-login
```
is not authenticated.

Why was the `/jetpack.authorize_iframe` allowed to use 3rd party cookies and the REST proxy iframe was not? I'm not sure, but Firefox probably decided that with some heuristics. `wordpress.com` is a domain I previously visited as first party and logged in there. That can be taken into account. And that heuristic no longer applies for the second nested iframe.

**Comment by jeherve** [comment]
Thanks for the report. The iFramed connection flow is going away in the next release (see #19872), so this shouldn't be an issue for much longer.

**Comment by jsnajdr** [comment]
Thanks for the response. I should also note that when trying the same task in Safari, with ITP enabled and 3rd party cookies strictly blocked with no exceptions, the flow was different. No popup window with login, no iframes. There was a top-level navigation to wordpress.com/jetpack/connect and back and the connection proceeded flawlessly.

**Comment by jeherve** [comment]
> There was a top-level navigation to wordpress.com/jetpack/connect and back and the connection proceeded flawlessly.

Yes, this is the old flow that will become the default flow again in the next release.

With the current iFramed flow, we've tried to properly fallback to the non-iFramed flow when necessary (see #19037 for some more details / examples), but as you've experienced it's not perfect. This is one of the reasons the iFramed flow is going away.

**Comment by jsnajdr** [comment]
To confirm my understanding: it is the "in place flow" that will be removed and it will be removed despite the good results (17% increase in finished connections) reported in p1HpG7-8SU-p2?

**Comment by jeherve** [comment]
That's the one, yes. You can find out more about all this in p1HpG7-bJT-p2

**Comment by jeherve** [comment]
The in-place flow has now been removed, and this is no longer an issue. Closing.
